frontend/simple/js/backend/hapi.js

```diff
@@ -105,12 +109,17 @@ export class HapiNamespace extends TrustedNamespace {
   async register (name: string, value: string) {
     console.log(`registering name:`, name)
     let response = await request.post(`${process.env.API_URL}/name`).send({name, value})
+    store.dispatch('addUser', {username: name, identity: value})
     return response.body.data
   }
   async lookup (name: string) {
     console.log(`lookup name:`, name)
     // TODO: should `name` be encodeURI'd?
+    if (store.state.users[name]) { return store.state.users[name] }
     let response = await request.get(`${process.env.API_URL}/name/${name}`)
+    store.dispatch('addUser', {username: name, identity: response.body.data.value})
```

Comment:
Lookup shouldn't have side effects like this. It's not at all clear/intuitive that looking up something will result in a change to the state, so this function should be left unmodified. Plus remember these functions must behave in a generic way, so that they are consistent across backends.